Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM ocaml/opam:debian-10-ocaml-4.11
+
+LABEL maintainer="gracewgao"
+
+SHELL ["/bin/bash", "-c"]
+
+# updates environment variables
+ENV PYRE_BINARY='/home/opam/pyre-check/source/_build/default/main.exe'
+ENV PYRE_TYPESHED='/home/opam/pyre-check/stubs/typeshed/typeshed-master'
+ENV PYTHONPATH='/home/opam/pyre-check:$PYTHONPATH'
+ENV PATH='/home/opam/.opam/4.10.2/bin:/home/opam/pyre-check/scripts:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$PATH'
+
+RUN echo -e 'export PYRE_BINARY=/home/opam/pyre-check/source/_build/default/main.exe \n \
+    export PYRE_TYPESHED=/home/opam/pyre-check/stubs/typeshed/typeshed-master \n \
+    export PYTHONPATH=/home/opam/pyre-check:$PYTHONPATH \n \
+    export PATH=/home/opam/pyre-check/scripts:$PATH' >> ~/.bashrc
+
+# installs system dependencies
+RUN sudo apt update && sudo apt upgrade -y \
+    && sudo apt install python3 python3-pip pkg-config -y \
+    && sudo apt-get install libsqlite3-dev build-essential -y \ 
+    && opam switch create 4.10.2
+
+# downloads and installs pyre-check
+RUN git clone https://github.com/facebook/pyre-check \
+    && cd pyre-check \
+    && ./scripts/setup.sh --local
+    
+RUN cd pyre-check/source \
+    && make \
+    && make test
+
+# creates scripts for python alias and pyre-check
+RUN sudo touch /usr/bin/python \
+    && sudo chmod a+w+x /usr/bin/python \
+    && sudo echo -e '#!/bin/bash\n \
+    python3 "$@" ' > /usr/bin/python \
+    && sudo echo -e '#!/bin/bash\n \
+    python -m client.pyre "$@" \n' >> /home/opam/pyre-check/scripts/pyre \
+    && sudo chmod +x /home/opam/pyre-check/scripts/pyre
+
+# runs python tests
+RUN cd /home/opam/pyre-check \ 
+    && pip3 install -r requirements.txt \
```

Comment:
Same comment as on the `apt` step. Installing dependencies should probably go earlier

Reply:
We can't install the dependencies until the requirements.txt inside the repo is cloned, so it would still have to be below that? I put them where they are now because the pip dependencies aren't really used for anything other than the Python tests, but I can change that if you want.

Reply:
Generally you want expensive, infrequently changed steps higher in the file. So I'd say the correct order would be
1. apt installs
2. bring in requirements.txt
3. pip install
4. bring in typeshed and unzip
4. bring in rest of repo
5. build

Doing it that way means you never have to re-run the install or unzip steps, even if you're actively editing and changing code in the repo and re-launching the container

Reply:
Okay, that makes sense - I'll change that now!